src/client/datascience/jupyter/kernels/kernelExecution.ts

```diff
@@ -332,6 +332,10 @@ export class KernelExecution implements IDisposable {
         );
     }
     private createQueuedCellExecution(cell: NotebookCell) {
+        const existingCellExecution = this.cellExecutions.get(cell);
```

Comment:
Good catch. 

Seems like a fine fix for now, but I did have a bit of an architectural question here. Should either the execution factory, or something that contains the execution factory be wrapping both the existing check and the create function?

Bit of a tricky thing at times with these small services. Someone else working in this class could call executionFactory.create without the check that you've added here, and again mess things up. If it's the case that we should never, ever create a cell execution when one already exists, then it might make sense to have another layer here or a structural change that would prevent that from happening.

Reply:
Yeah maybe this could be refactored when you make the other refactor change. Make it impossible to have more than one execution for the same cell.